### PR TITLE
pythonPackages.*: switch many packages to disabledTestPaths

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Motivation for this change
Replaces #115303

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/kubernetes/default.nix [hunk 1]
```diff
@@ -1,34 +1,32 @@
...
-  checkInputs = [ isort coverage pytest mock sphinx autopep8 pep8 codecov recommonmark nose ];
+  # prePatch = ''
+  #   sed -e 's/sphinx>=1.2.1,!=1.3b1,<1.4 # BSD/sphinx/' -i test-requirements.txt
+
+  #   # This is used to randomize tests, which is not reproducible. Drop it.
+  #   sed -e '/randomize/d' -i test-requirements.txt
+  # ''
+  # # This is a python2 and python3.2 only requiremet since it is a backport of a python-3.3 api.
+  # + lib.optionalString (pythonAtLeast "3.3")
+  # "sed -e '/ipaddress/d' -i requirements.txt";
```
Comment (risicle) [inline pkgs/development/python-modules/kubernetes/default.nix:22]: What's the story with these? Do you think they're now redundant?

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -4395,7 +4395,9 @@ in {
 
   nbconflux = callPackage ../development/python-modules/nbconflux { };
 
-  nbconvert = callPackage ../development/python-modules/nbconvert { };
+  nbconvert = callPackage ../development/python-modules/nbconvert {
+    nbclient = self.nbclient.override { doCheck = false; };
```
Comment (risicle) [inline pkgs/top-level/python-packages.nix:4399]: Could do with the "avoid recursion" comment left here but :shrug:

File: pkgs/tools/networking/linkchecker/default.nix [hunk 3]
```diff
@@ -26,23 +26,28 @@ buildPythonApplication rec {
...
 
-  postPatch = ''
-    sed -i 's/^requests.*$/requests>=2.2/' requirements.txt
-    sed -i "s/'request.*'/'requests >= 2.2'/" setup.py
+  # postPatch = ''
+  #   sed -i 's/^requests.*$/requests>=2.2/' requirements.txt
+  #   sed -i "s/'request.*'/'requests >= 2.2'/" setup.py
+  # '';
+
+  # network tests fails on darwin
```
Comment (risicle) [inline pkgs/tools/networking/linkchecker/default.nix:37]: They fail to import?

File: pkgs/development/python-modules/smbus-cffi/default.nix [hunk 4]
```diff
@@ -26,16 +26,18 @@ buildPythonPackage rec {
 
   propagatedBuildInputs = [ cffi ];
 
+  checkInputs = [ pytestCheckHook pyserial ];
+
   installCheckPhase = ''
     # we want to import the installed module that also contains the compiled library
-    rm -rf smbus
+    rm -r smbu
```
Comment (risicle) [inline pkgs/development/python-modules/smbus-cffi/default.nix:33]: Mistake?

File: pkgs/development/python-modules/kubernetes/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/kubernetes/default.nix:22]: most likely if the package isn't used in python 2 anymore.

File: pkgs/development/python-modules/smbus-cffi/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/smbus-cffi/default.nix:33]: No, ``rm -f`` should be avoided because it does not fail ever.

Comment (risicle) [reply pkgs/development/python-modules/smbus-cffi/default.nix:33]: No, I mean `smbu` -> `smbus`

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (doronbehar) [reply pkgs/top-level/python-packages.nix:4399]: I think `nbclient` needs to have it's tests disabled always, as discussed in https://github.com/NixOS/nixpkgs/pull/115303#discussion_r589514942 .

**Comment by doronbehar** [comment]
There are many good changes here @SuperSandro2000 , and some merge conflicts now. Perhaps the amount of packages changed can be reduced to such an amount that the PR won't require using `staging` as the base branch.

**Pull request closed without merging**